### PR TITLE
Feature/jenkins plugin

### PR DESCRIPTION
**Pull request by shudgston**
* Adds a jenkins client class for interacting with the jenkins HTTP API.
* Adds setup steps to prompt user for jenkins authentication (stores in .gitconfig)
* Adds new `cirrus release trigger` command which performs the steps of `cirrus release new` and initiates a jenkins build job passing the new release/version number.

### CODE REVIEW

File: cirrus.conf [hunk 1]
```diff
@@ -28,3 +28,9 @@ threshold = 10
 [test-default]
 where = tests/
 
+[build-server]
```
Comment (evansde77) [inline cirrus.conf]: given we arent actually building cirrus with jenkins, is this config just for testing?

File: src/cirrus/release.py [hunk 2]
```diff
@@ -257,7 +259,46 @@ def new_release(opts):
...
+    _trigger_release_
+
+    Alias for "git cirrus release new --micro/minor/major.
+    - Run the "release new" command
+    - Capture the new version string
+    - Pass new version number to external build server
+    """
+    new_version, release_level = new_release(opts)
+    config = load_configuration()
+    build_server = config['build-server']['name']
```
Comment (evansde77) [inline src/cirrus/release.py]: check this is present and throw with helpful "update your conf..." blurb?

**Comment by evansde77** [comment]
Couple comments, but broadly LGTM

File: cirrus.conf (hunk 1 above)
Comment (shudgston) [inline cirrus.conf]: Yes I think this can be removed

**Pull request merged**